File: focus_todo_computer/focus-todo/src/sync/syncTransportRegistry.ts
import type { CanonicalTask } from './canonicalTaskMapper'
import { FakeSyncTransport } from './fakeSyncTransport'
import { SupabaseSyncTransport } from './supabaseSyncTransport'
import type { SyncTransport } from './syncTransport'

export type SyncTransportMode = 'fake' | 'supabase'

export class SyncTransportRegistry {
  private mode: SyncTransportMode = 'fake'
  private readonly fakeTransport = new FakeSyncTransport()
  private readonly supabaseTransport = new SupabaseSyncTransport()

  getMode(): SyncTransportMode {
    return this.mode
  }

  setMode(mode: SyncTransportMode): void {
    this.mode = mode
  }

  getTransport(): SyncTransport {
    return this.mode === 'supabase'
      ? this.supabaseTransport
      : this.fakeTransport
  }

  getRemoteSnapshot(): CanonicalTask[] {
    return this.mode === 'fake' ? this.fakeTransport.snapshot() : []
  }
}

File: focus_todo_mobile/lib/sync/sync_transport_registry.dart
import 'canonical_task_mapper.dart';
import 'fake_sync_transport.dart';
import 'supabase_sync_transport.dart';
import 'sync_transport.dart';

enum SyncTransportMode { fake, supabase }

class SyncTransportRegistry {
  final FakeSyncTransport _fakeTransport = FakeSyncTransport();
  final SupabaseSyncTransport _supabaseTransport = SupabaseSyncTransport();
  SyncTransportMode _mode = SyncTransportMode.fake;

  SyncTransportMode getMode() => _mode;

  void setMode(SyncTransportMode mode) {
    _mode = mode;
  }

  SyncTransport getTransport() {
    return _mode == SyncTransportMode.supabase
        ? _supabaseTransport
        : _fakeTransport;
  }

  List<CanonicalTask> getRemoteSnapshot() {
    return _mode == SyncTransportMode.fake
        ? _fakeTransport.snapshot()
        : const <CanonicalTask>[];
  }
}
